packages/@aws-cdk/aws-ec2/lib/nat.ts

```diff
@@ -0,0 +1,218 @@
+import iam = require('@aws-cdk/aws-iam');
+import { Instance } from './instance';
+import { InstanceType } from './instance-types';
+import { IMachineImage, LookupMachineImage } from "./machine-image";
+import { Port } from './port';
+import { SecurityGroup } from './security-group';
+import { PrivateSubnet, PublicSubnet, RouterType, Vpc } from './vpc';
+
+/**
+ * NAT providers
+ *
+ * Determines what type of NAT provider to create, either NAT gateways or NAT
+ * instance.
+ *
+ * @experimental
+ */
+export abstract class NatProvider {
+  /**
+   * Use NAT Gateways to provide NAT services for your VPC
+   *
+   * NAT gateways are managed by AWS.
+   *
+   * @see https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-gateway.html
+   */
+  public static gateway(): NatProvider {
+    return new NatGateway();
+  }
+
+  /**
+   * Use NAT instances to provide NAT services for your VPC
+   *
+   * NAT instances are managed by you, but in return allow more configuration.
+   *
+   * Be aware that instances created using this provider will not be
+   * automatically replaced if they are stopped for any reason. You should implement
+   * your own NatProvider based on AutoScaling groups if you need that.
+   *
+   * @see https://docs.aws.amazon.com/vpc/latest/userguide/VPC_NAT_Instance.html
+   */
+  public static instance(props: NatInstanceProps): NatProvider {
+    return new NatInstance(props);
+  }
+
+  /**
+   * Called by the VPC to configure NAT
+   */
+  public abstract configureNat(options: ConfigureNatOptions): void;
+}
+
+/**
+ * Options passed by the VPC when NAT needs to be configured
+ *
+ * @experimental
+ */
+export interface ConfigureNatOptions {
+  /**
+   * The VPC we're configuring NAT for
+   */
+  readonly vpc: Vpc;
+
+  /**
+   * The public subnets where the NAT providers need to be placed
+   */
+  readonly natSubnets: PublicSubnet[];
+
+  /**
+   * The private subnets that need to route through the NAT providers.
+   *
+   * There may be more private subnets than public subnets with NAT providers.
+   */
+  readonly privateSubnets: PrivateSubnet[];
+}
+
+/**
+ * Properties for a NAT instance
+ *
+ * @experimental
+ */
+export interface NatInstanceProps {
+  /**
+   * The machine image (AMI) to use
+   *
+   * By default, will do an AMI lookup for the latest NAT instance image.
+   *
+   * If you have a specific AMI ID you want to use, pass a `GenericLinuxImage`. For example:
+   *
+   * ```ts
+   * NatProvider.instance({
+   *   instanceType: new InstanceType('t3.micro'),
+   *   machineImage: new GenericLinuxImage({
+   *     'us-east-2': 'ami-0f9c61b5a562a16af'
+   *   })
+   * })
+   * ```
+   *
+   * @default - Latest NAT instance image
+   */
+  readonly machineImage?: IMachineImage;
+
+  /**
+   * Instance type of the NAT instance
+   */
+  readonly instanceType: InstanceType;
```

Comment:
default?

Reply:
It's not optional.

Reply:
Let's make it optional and pick a sensible default, no?